presto-main/src/main/java/io/prestosql/operator/ExchangeOperator.java

```diff
@@ -72,7 +72,7 @@ public SourceOperator createOperator(DriverContext driverContext)
             checkState(!closed, "Factory is already closed");
             OperatorContext operatorContext = driverContext.addOperatorContext(operatorId, sourceId, ExchangeOperator.class.getSimpleName());
             if (exchangeClient == null) {
-                exchangeClient = exchangeClientSupplier.get(driverContext.getPipelineContext().localSystemMemoryContext());
+                exchangeClient = exchangeClientSupplier.get(operatorContext.localSystemMemoryContext());
```

Comment:
I don't think you can do this because the client is shared by all operators.  So if the first operator finishes before the others, the memory usage will drop to zero.

Reply:
Indeed. Didn't notice that. Closing this PR then.

I wonder if it's beneficial to have some pipeline level `OperatorContext`.